src/_pytest/fixtures.py

```diff
@@ -593,6 +595,39 @@ def _get_active_fixturedef(
             raise FixtureLookupError(argname, self)
         fixturedef = fixturedefs[index]
 
+        # Check for attempted use of an async fixture by a sync test
+        # `self.scope` here is not the scope of the requested fixture, but the scope of
+        # the requester.
+        if (
+            self.scope == "function"
+            and not inspect.iscoroutinefunction(self._pyfuncitem.obj)
+            and (
+                inspect.iscoroutinefunction(fixturedef.func)
+                or inspect.isasyncgenfunction(fixturedef.func)
+            )
+        ):
```

Comment:
I'm not *100%* sure this is the place to do the check, and am also not 100% it can't be triggered by a sync fixture requesting an async fixture. But for the latter I think it's covered by the `self.scope == "function"` check, where if `self` is a fixture requesting another fixture it's because it's higher-scoped.
So while this appears to function robustly, it might be making somewhat sketchy assumptions.

Reply:
I think this is correct, and I also see a test for this, so I guess we are good.